tests/test_cookiecutter_python_library.py

```diff
@@ -109,7 +112,5 @@ def test_setup_py(options_baked):
 def test_tests_and_quality(options_baked):
     """
     Run generated tests and quality checks.
-
-    This also tests `make upgrade` by necessity, since it needs the requirements files.
     """
-    run_in_virtualenv('make upgrade requirements test-all')
```

Comment:
I think we should continue doing the testing inside a separate virtualenv. Otherwise we'll get contamination between tests, and not be able to detect things like missing dependencies.

Probably the best thing to do would be to 1) run `make upgrade` as part of the post-gen hooks so that it's already in place for developers and unit tests alike, and 2) adjust `run_in_virtualenv` to accept a sequence of commands to run.

(Maybe introduce a uniform `test-requirements` make target as well? I do like the change to just installing test deps...)

Reply:
For this pull request, I can leave the `make upgrade` as part of the module-scoped cookie baking, but put back the virtualenv run for running the new repo's tests.